SSA/Projects/RISCV64/Semantics.lean

```diff
@@ -53,21 +124,68 @@ def RTYPEW_pure64_RISCV_SUBW (rs2_val : BitVec 64) (rs1_val : BitVec 64) : BitVe
   BitVec.signExtend 64 (BitVec.sub rs1_val32 rs2_val32)
 
 def RTYPEW_pure64_RISCV_SLLW (rs2_val : BitVec 64) (rs1_val : BitVec 64) : BitVec 64 :=
+  let rs1_val32 := BitVec.extractLsb' 0 32 rs1_val;
+  let rs2_val32 := BitVec.extractLsb' 0 32 rs2_val;
+  let shamt := BitVec.extractLsb' 0 5 rs2_val32;
+  BitVec.signExtend 64 (BitVec.shiftLeft rs1_val32 shamt.toNat)
+
+def RTYPEW_pure64_RISCV_SLLW_bv (rs2_val : BitVec 64) (rs1_val : BitVec 64) : BitVec 64 :=
   let rs1_val32 := BitVec.extractLsb' 0 32 rs1_val;
   let rs2_val32 := BitVec.extractLsb' 0 32 rs2_val;
   let shamt := BitVec.extractLsb' 0 5 rs2_val32;
   BitVec.signExtend 64 (rs1_val32 <<< shamt)
 
+theorem RTYPEW_pure64_RISCV_SLLW_eq_RTYPEW_pure64_RISCV_SLLW_bv
+  (rs2_val : BitVec 64) (rs1_val : BitVec 64) :
+  RTYPEW_pure64_RISCV_SLLW rs2_val rs1_val
+  = RTYPEW_pure64_RISCV_SLLW_bv rs2_val rs1_val :=
+  by
+  unfold RTYPEW_pure64_RISCV_SLLW RTYPEW_pure64_RISCV_SLLW_bv
+  simp only [extractLsb'_eq_setWidth, Nat.reduceLeDiff, BitVec.setWidth_setWidth_of_le,
+    BitVec.toNat_setWidth, Nat.reducePow, BitVec.shiftLeft_eq, BitVec.shiftLeft_eq']
+
 def RTYPEW_pure64_RISCV_SRLW (rs2_val : BitVec 64) (rs1_val : BitVec 64) : BitVec 64 :=
+  let rs1_val32 := BitVec.extractLsb' 0 32 rs1_val;
+  let rs2_val32 := BitVec.extractLsb' 0 32 rs2_val;
+  let shamt := BitVec.extractLsb' 0 5 rs2_val32;
+  BitVec.signExtend 64 (BitVec.ushiftRight rs1_val32 shamt.toNat)
+
+def RTYPEW_pure64_RISCV_SRLW_bv (rs2_val : BitVec 64) (rs1_val : BitVec 64) : BitVec 64 :=
   let rs1_val32 := BitVec.extractLsb' 0 32 rs1_val;
   let rs2_val32 := BitVec.extractLsb' 0 32 rs2_val;
   let shamt := BitVec.extractLsb' 0 5 rs2_val32;
   BitVec.signExtend 64 (rs1_val32 >>> shamt)
 
-def RTYPEW_pure64_RISCV_SRAW (rs2_val : BitVec 64) (rs1_val : BitVec 64) : BitVec 64 :=
+theorem RTYPEW_pure64_RISCV_SLLW_eq_RTYPEW_pure64_RISCV_SRLW_bv
+  (rs2_val : BitVec 64) (rs1_val : BitVec 64) :
+  RTYPEW_pure64_RISCV_SRLW rs2_val rs1_val
+  = RTYPEW_pure64_RISCV_SRLW_bv rs2_val rs1_val := by
+  unfold RTYPEW_pure64_RISCV_SRLW RTYPEW_pure64_RISCV_SRLW_bv
+  simp only [extractLsb'_eq_setWidth, Nat.reduceLeDiff, BitVec.setWidth_setWidth_of_le,
+    BitVec.toNat_setWidth, Nat.reducePow, BitVec.ushiftRight_eq, BitVec.ushiftRight_eq']
```

Comment:
A terminal simp does not need to be `simp only`. Drop the `only` and see if you can drop some of the arguments to this `simp` call.